### DISCUSSION TITLE
PDF OCR function turned on but not working well.

### DISCUSSION THREAD

**Discussion by edb616321** [open-webui/open-webui#3448] [General]
llama3/ollama/openwebui. nvidia 4090 and a high end computer. When I upload PDF files it stores them. When I use the hashtag '#' to use 'all documents' or individual documents, it seems to comprehend about 30-50% of the document and maybe 10% of the embedded  images that should be OCR processed. Can I choose a more powerful OCR system or tweak the existing one. The document processing seems remedial.  copy the same document to chatgpt4 and ask the same question and get a extremely improved answer. Do  I have to build a Function/ Tool for this process? Any help would be much appreciated.

**Comment by Otts86** [comment]
I'm having the same issue and would like to see this functionality dramatically improved.

**Reply by justinh-rahb** [reply]
> I'm having the same issue and would like to see this functionality dramatically improved.

Enable Apache Tika for **Content Extraction** in **Admin Settings > Documents**.

---
### Running Tika

Add a container using `docker run`:
```bash
docker run -d --name tika -p 9998:9998 --restart unless-stopped apache/tika:latest-full
```

Or you can add this `docker-compose.tika.yaml` to your Compose stack:
```yml
services:
  tika:
    image: apache/tika:latest-full
    container_name: tika
    restart: unless-stopped
```
```bash
docker compose -f docker-compose.yaml -f docker-compose.tika.yaml up -d
```

**Reply by mdlmarkham** [reply]
I've been running Tika for a day - so far it's significantly improved the way that PDFs are processed.

**Reply by Otts86** [reply]
For those of you with experience running Tika - what flavor would provide the best extraction for PDF files - NER, CustomOCR, Grobid, or Vision?  I've been running NER but am not sure I'm getting the most out of it.  I read somewhere about newer models beyond 1.5.  Any help appreciated.

**Reply by pbpippen** [reply]
awesome! thanks for the tip @justinh-rahb

**Reply by lrud** [reply]
this certainly worked thank you! my only issue is when adding <1 file/pdf into the context window, the LLM has issues. Could this be hardware limitatinos?

**Comment by twythebest** [comment]
why I add tika，still like not use tika:a pdf file cannot see itself?!!!

**Comment by enving** [comment]
make sure you dont forget to put it in the same network in the docker compose as openwebui also is... (I did that :D )
